### ISSUE TITLE
Hosts defined in container are not recognized

### ISSUE THREAD

**Issue by srigi** [michaelshobbs/dokku-app-predeploy-tasks#11]
As I understand this script runs outside of the container in the context of server machine.
My database configuration for container contains special hostname configured via `docker run --link`.
When I run migrations, this special hostname is not resolved because it is configured inside container's `/etc/hosts`.

Any ideas?

**Comment by michaelshobbs** [comment]
This plugin runs inside of a container as well. Can you provide a gist of a failed deployment with `dokku trace on`?

**Comment by srigi** [comment]
OK, so I run `dokku 0.3.22` with latest plugins `Flink/dokku-psql-single-container` and yours `michaelshobbs/dokku-app-predeploy-tasks`. Minimal project is like this:

**composer.json**
```json
{
  "name": "predeploy-test",
  "require": {
    "php": ">= 5.4.0"
  }
}
```
**Procfile**
```
web: vendor/bin/heroku-php-apache2 www/
test1: php -r 'var_dump(gethostbyname("google.com"));'
test2: php -r 'var_dump(gethostbyname("postgres"));'
```
**SCALE**
```
test1=0
test2=0
```
**www/index.php**
```php
<pre>
<?php

var_dump(gethostbyname("google.sk"));
var_dump(gethostbyname("postgres"));
```

Then I created both empty dokku app & psql database
```
# dokku apps:create predeploy
# dokku psql:create predeploy
```
psql plugin automatically links containers with the same name

```
# dokku config predeploy
=====> predeploy config vars
DATABASE_URL:   postgresql://predeploy:c0475f1f96e9935f40dce5bc43a14e24174039dd0d69264631b4847fb9ebd932@postgres:5432/predeploy
DB_HOST:        postgres
DB_NAME:        predeploy
DB_PASS:        c0475f1f96e9935f40dce5bc43a14e24174039dd0d69264631b4847fb9ebd932
DB_PORT:        5432
DB_TYPE:        postgresql
DB_USER:        predeploy
POSTGRESQL_URL: postgresql://predeploy:c0475f1f96e9935f40dce5bc43a14e24174039dd0d69264631b4847fb9ebd932@postgres:5432/predeploy
```
Please note the `DB_HOST` env variable. After deploying this app ([log with trace on](http://pastebin.com/cqtS91fA)) I get this output from **dokku-app-predeploy-tasks** plugin:
```
# cat /tmp/dokku-predeploy-tasks-logs/predeploy-tasks-2015-07-28-17-46-22.log
restoring installation cache...
=====> running task test1
string(13) "195.80.179.42"
=====> running task test2
string(8) "postgres"
removing installation cache...
```

The thing is, `postgres` hostname should be resolved as well, because `psql-single-container` is "*fiddling*" with `/etc/hosts` during every phase of build (build, deploy & run). When you run abovementioned task from within running container, it resolve w/o problem.

**Comment by michaelshobbs** [comment]
Got it. Just as I suspected. We're not getting `$DOCKER_ARGS` correctly. See lines 1668-1669. I'll get a patch going here shortly. Thanks for the report.

**Comment by michaelshobbs** [comment]
Also, thanks a ton for the detailed bug report. Super helpful!

**Comment by michaelshobbs** [comment]
I think this will be fixed by #12. Give it a shot and let me know.
